src/history/hs.c

```diff
@@ -842,8 +842,7 @@ __wt_hs_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t bt
      * history store as opposed to comparing the embedded data store key since the ordering is not
      * guaranteed to be the same.
      */
-    cursor->set_key(
-      cursor, btree_id, key, timestamp != WT_TS_NONE ? timestamp : WT_TS_MAX, UINT64_MAX);
+    cursor->set_key(cursor, btree_id, key, timestamp, UINT64_MAX);
```

Comment:
I moved this part out to the caller since this function is now used in multiple places not just in reading history store records. I think this function should just search the corresponding records based on the input from the caller.

Reply:
revert this